File: CS/Spelling.Win/SpellCheckerViewController.cs
using System;
using System.Windows.Forms;
using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.XtraSpellChecker;
using DevExpress.ExpressApp.Actions;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.ExpressApp.Win.Editors;

namespace Spelling.Win {
    public class SpellCheckerViewController : ViewController {
        private SpellChecker spellCheckerCore;
        private SimpleAction checkActionCore;
        public SpellCheckerViewController() {
            checkActionCore = new SimpleAction(this, "Check", PredefinedCategory.RecordEdit);
            checkActionCore.Caption = checkActionCore.ToolTip = "Check Spelling";
            checkActionCore.Category = "RecordEdit";
            checkActionCore.ImageName = "BO_Task";
            checkActionCore.TargetViewType = ViewType.DetailView;
            checkActionCore.Execute += checkActionCore_Execute;
        }
        protected override void OnViewControlsCreated() {
            base.OnViewControlsCreated();
            SetupSpellChecker();
            SetupGridView();
        }
        protected override void OnDeactivated() {
            checkActionCore.Execute -= checkActionCore_Execute;
            spellCheckerCore = null;
            checkActionCore = null;
            base.OnDeactivated();
        }
        protected virtual void SetupSpellChecker() {
            spellCheckerCore = new SpellChecker();
            if (spellCheckerCore.ParentContainer == null) {
                spellCheckerCore.ParentContainer = (Control)View.Control;
            }
            SpellCheckerInitializer.Initialize(spellCheckerCore);
        }
        protected virtual void SetupGridView() {
            if (View is DevExpress.ExpressApp.ListView) {
                DevExpress.ExpressApp.ListView lv = (DevExpress.ExpressApp.ListView)View;
                GridListEditor gridListEditor = lv.Editor as GridListEditor;
                if (gridListEditor != null) {
                    gridListEditor.GridView.ShownEditor -= GridView_ShownEditor;
                    gridListEditor.GridView.ShownEditor += GridView_ShownEditor;
                    gridListEditor.GridView.Disposed += GridView_Disposed;
                }
            }
        }
        void GridView_Disposed(object sender, EventArgs e) {
            GridView gv = (GridView)sender;
            gv.ShownEditor -= GridView_ShownEditor;
            gv.Disposed -= GridView_Disposed;
        }
        private void GridView_ShownEditor(object sender, EventArgs e) {
            Control activeControl = ((GridView)sender).ActiveEditor;
            spellCheckerCore.SetShowSpellCheckMenu(activeControl, true);
            if (spellCheckerCore.SpellCheckMode == SpellCheckMode.AsYouType) {
                //spellCheckerCore.Check(activeControl);
            }
        }
        private void checkActionCore_Execute(object sender, SimpleActionExecuteEventArgs e) {
            spellCheckerCore.CheckContainer();
        }
        public SpellChecker SpellChecker {
            get { return spellCheckerCore; }
        }
        public SimpleAction CheckAction {
            get { return checkActionCore; }
        }
    }
}

File: VB/Spelling.Win/SpellCheckerViewController.vb
Imports System
Imports System.Windows.Forms
Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.XtraSpellChecker
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.ExpressApp.Win.Editors

Namespace Spelling.Win
    Public Class SpellCheckerViewController
        Inherits ViewController

        Private spellCheckerCore As SpellChecker
        Private checkActionCore As SimpleAction
        Public Sub New()
            checkActionCore = New SimpleAction(Me, "Check", PredefinedCategory.RecordEdit)
            checkActionCore.ToolTip = "Check Spelling"
            checkActionCore.Caption = checkActionCore.ToolTip
            checkActionCore.Category = "RecordEdit"
            checkActionCore.ImageName = "BO_Task"
            checkActionCore.TargetViewType = ViewType.DetailView
            AddHandler checkActionCore.Execute, AddressOf checkActionCore_Execute
        End Sub
        Protected Overrides Sub OnViewControlsCreated()
            MyBase.OnViewControlsCreated()
            SetupSpellChecker()
            SetupGridView()
        End Sub
        Protected Overrides Sub OnDeactivated()
            RemoveHandler checkActionCore.Execute, AddressOf checkActionCore_Execute
            spellCheckerCore = Nothing
            checkActionCore = Nothing
            MyBase.OnDeactivated()
        End Sub
        Protected Overridable Sub SetupSpellChecker()
            spellCheckerCore = New SpellChecker()
            If spellCheckerCore.ParentContainer Is Nothing Then
                spellCheckerCore.ParentContainer = CType(View.Control, Control)
            End If
            SpellCheckerInitializer.Initialize(spellCheckerCore)
        End Sub
        Protected Overridable Sub SetupGridView()
            If TypeOf View Is DevExpress.ExpressApp.ListView Then
                Dim lv As DevExpress.ExpressApp.ListView = CType(View, DevExpress.ExpressApp.ListView)
                Dim gridListEditor As GridListEditor = TryCast(lv.Editor, GridListEditor)
                If gridListEditor IsNot Nothing Then
                    RemoveHandler gridListEditor.GridView.ShownEditor, AddressOf GridView_ShownEditor
                    AddHandler gridListEditor.GridView.ShownEditor, AddressOf GridView_ShownEditor
                    AddHandler gridListEditor.GridView.Disposed, AddressOf GridView_Disposed
                End If
            End If
        End Sub
        Private Sub GridView_Disposed(ByVal sender As Object, ByVal e As EventArgs)
            Dim gv As GridView = DirectCast(sender, GridView)
            RemoveHandler gv.ShownEditor, AddressOf GridView_ShownEditor
            RemoveHandler gv.Disposed, AddressOf GridView_Disposed
        End Sub
        Private Sub GridView_ShownEditor(ByVal sender As Object, ByVal e As EventArgs)
            Dim activeControl As Control = DirectCast(sender, GridView).ActiveEditor
            spellCheckerCore.SetShowSpellCheckMenu(activeControl, True)
            If spellCheckerCore.SpellCheckMode = SpellCheckMode.AsYouType Then
                'spellCheckerCore.Check(activeControl);
            End If
        End Sub
        Private Sub checkActionCore_Execute(ByVal sender As Object, ByVal e As SimpleActionExecuteEventArgs)
            spellCheckerCore.CheckContainer()
        End Sub
        Public ReadOnly Property SpellChecker() As SpellChecker
            Get
                Return spellCheckerCore
            End Get
        End Property
        Public ReadOnly Property CheckAction() As SimpleAction
            Get
                Return checkActionCore
            End Get
        End Property
    End Class
End Namespace